example/lib/screen/nav_bar/nav_bar_reader_node_screen.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+import 'base_nav_bar_screen.dart';
+
+class NavBarReaderModeScreen extends BaseNavBarScreen {
+  static const routeName = '/navBarReaderModeScreen';
+
+  const NavBarReaderModeScreen({Key? key}) : super(key: key);
+
+  @override
+  _NavBarHomeScreenState createState() => _NavBarHomeScreenState();
+}
+
+class _NavBarHomeScreenState
+    extends BaseNavBarScreenState<NavBarReaderModeScreen>
+    with NavBarConfigMixin {
+  @override
+  String get screenTitle => 'Reader mode screen';
+
+  @override
+  Color get screenBgColor => Colors.deepPurpleAccent;
+
+  bool? isLiked;
+
+  @override
+  NavBarConfig get navBarConfig {
+    final goBack = NavBarItemIconButton(
+      svgIconPath: linden.assets.icons.arrowLeft,
+      isSelected: false,
+      onPressed: () {
+        Navigator.of(context).pop();
+      },
+      key: back,
+    );
+
+    final like = NavBarItemIconButton(
+      svgIconPath: isLiked == true
+          ? linden.assets.icons.thumbsUpActive
+          : linden.assets.icons.thumbsUp,
+      isSelected: false,
+      onPressed: () {
+        if (isLiked == null || isLiked == false) {
+          isLiked = true;
+        } else if (isLiked == true) {
+          isLiked = null;
+        }
+        NavBarContainer.updateNavBar(context);
+      },
+      key: keyLike,
+    );
+    final dislike = NavBarItemIconButton(
+      svgIconPath: isLiked == false
+          ? linden.assets.icons.thumbsDownActive
+          : linden.assets.icons.thumbsDown,
+      isSelected: false,
+      onPressed: () {
+        if (isLiked == null || isLiked == true) {
+          isLiked = false;
+        } else if (isLiked == false) {
+          isLiked = null;
+        }
+        NavBarContainer.updateNavBar(context);
+      },
+      key: keyDisLike,
```

Comment:
Ok this is I guess the tricky part, how to nicely represent State here. A suggestion is the following.
The `NavBarItem` adds a generic State object that can be anything from boolean to a complex object.

```dart
class NavBarItem<S> {
  S get state => ...
  set state(update) =>  updateAllMethods(update)

   ...  
}

navBarItemIconButton(
      svgIconPath: (state) => state.pressed 
          ? linden.assets.icons.thumbsDownActive
          : linden.assets.icons.thumbsDown,
      initialState: ButtonState(isPressed : false),
      onPressed: (prevState) {
        if (isLiked == null || isLiked == true) {
          return ButtonState(isPressed : true)
        } else if (isLiked == false) {
         return ButtonState(isPressed : false),
        }
      }
```
      
Now it would be possible to compare the state and update always the icon or text or whatever methods . And you don't have to drag the state outside in other classes.

WDYT?

Reply:
So far we decided to left it as it is right now, and maybe later come again to this idea